vendor/bat-native-ledger/BUILD.gn

```diff
@@ -14,14 +14,20 @@ if (is_android) {
 dep_base = rebase_path("../..", "//")
 
 config("external_config") {
-  visibility = [ ":*" ]
+  visibility = [
+    ":*",
+    "//brave/vendor/bat-native-ads:ads",
```

Comment:
these visibility lists exist to prevent people from doing this. This config is accessed as a public config on `headers`

Reply:
we probably want to split these defines out into a separate config that is allowed from `//brave/vendor/bat-native-ads:ads`

Reply:
they really shouldn't be in a public config anyway because `GEMINI_OAUTH_STAGING_URL` and `BITFLYER_STAGING_URL` aren't used outside of bat-native-ledger. The strictly internal ones above should go in the internal config and the new ones should go in a new config that has the `:*` and `//brave/vendor/bat-native-ads:ads`

Reply:
OK, I think I got misled a bit by the name `external_config`. Moved these into `endpoints_config`.